src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -43,14 +43,21 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
         public async Task<ExecutionResult> InitializeAsync() {
             try {
                 if (!Session.IsHostRunning) {
-                    await Session.StartHostAsync(new RHostStartupInfo {
+                    var startupInfo = new RHostStartupInfo {
                         Name = "REPL",
                         RBasePath = _settings.RBasePath,
                         RHostCommandLineArguments = _settings.RCommandLineArguments,
                         CranMirrorName = _settings.CranMirror,
                         CodePage = _settings.RCodePage,
-                        WorkingDirectory = _settings.WorkingDirectory
-                    }, new RSessionCallback(CurrentWindow, Session, _settings, _coreShell));
+                        WorkingDirectory = _settings.WorkingDirectory,
+                    };
+
+                    IInteractiveWindowVisualComponent visualComponent;
```

Comment:
`RInteractiveEvaluator` already has access to the `InteractiveWindow` through `CurrentWindow` property, so the whole `VisualElement_SizeChanged` can be implemented directly inside `RInteractiveEvaluator` itself. I know that this class pretends to be on the lower level than ViewModel, but `CurrentWindow` is defined by `IInteractiveEvaluator` itself, I don't think there is a reason to spread this logic between 3 classes.